pytorch_lightning/trainer/trainer.py

```diff
@@ -1177,16 +1178,18 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
             )
         return ckpt_path
 
-    def _call_setup_hook(self) -> None:
+    def _call_setup_model_hook(self, model: "pl.LightningModule") -> None:
         fn = self.state.fn._setup_fn
-
         self.accelerator.barrier("pre_setup")
+        self.setup(model, stage=fn)
```

Comment:
nit: maybe reflect in the method name that this also invokes the trainer `setup` hooks?